localstack/services/route53/provider.py

```diff
@@ -0,0 +1,145 @@
+        region_details = Route53Backend.get()
+        # TODO: handle NoSuchHostedZone and ConflictingDomainExist
+        zone_details = region_details.vpc_hosted_zone_associations.get(hosted_zone_id) or []
+        hosted_zone_association = HostedZoneAssociation(
+            hosted_zone_id=hosted_zone_id,
+            id=short_uid(),
+            vpc=vpc,
+            status=ChangeStatus.INSYNC,
+            submitted_at=datetime.now(),
+        )
+        zone_details.append(hosted_zone_association)
+        vpc_id = vpc.get("VPCId")
+        # update VPC info in hosted zone moto object - fixes required after https://github.com/spulec/moto/pull/4786
+        hosted_zone = route53_backend.zones.get(hosted_zone_id)
+        if not getattr(hosted_zone, "vpcid", None):
+            hosted_zone.vpcid = vpc_id
+        if not getattr(hosted_zone, "vpcregion", None):
+            hosted_zone.vpcregion = aws_stack.get_region()
+
+        region_details.vpc_hosted_zone_associations[hosted_zone_id] = zone_details
+        return AssociateVPCWithHostedZoneResponse(
+            ChangeInfo=ChangeInfo(
+                Id=short_uid(), Status=ChangeStatus.INSYNC, SubmittedAt=datetime.now()
+            )
+        )
+
+    def disassociate_vpc_from_hosted_zone(
+        self,
+        context: RequestContext,
+        hosted_zone_id: ResourceId,
+        vpc: VPC,
+        comment: DisassociateVPCComment = None,
+    ) -> DisassociateVPCFromHostedZoneResponse:
+        region_details = Route53Backend.get()
+        zone_details = region_details.vpc_hosted_zone_associations.get(hosted_zone_id) or []
+
+        def _match(_zone: HostedZoneAssociation):
+            return _zone.hosted_zone_id == hosted_zone_id and _zone.vpc.get("VPCId") == vpc.get(
+                "VPCId"
+            )
+
+        matching_hosted_zones = [z for z in zone_details if _match(z)]
+        not_matching_zones = [z for z in zone_details if not _match(z)]
+
+        if not matching_hosted_zones:
+            raise VPCAssociationNotFound()
+
+        region_details.vpc_hosted_zone_associations[hosted_zone_id] = not_matching_zones
+        return DisassociateVPCFromHostedZoneResponse(
+            ChangeInfo=ChangeInfo(
+                Id=matching_hosted_zones[0].id,
+                Status=matching_hosted_zones[0].status,
+                SubmittedAt=matching_hosted_zones[0].submitted_at,
+            )
+        )
+
+
+def get_or_delete_health_check(request, full_url, headers):
+    parsed_url = urlparse(full_url)
+    parts = parsed_url.path.strip("/").split("/")
+    health_check_id = parts[-1]
+    if request.method == "GET":
+        health_check = route53_backend.health_checks.get(health_check_id)
+        if not health_check:
+            return 404, {}, ""
+        result = """<GetHealthCheckResponse xmlns="{}">{}</GetHealthCheckResponse>"""
+        result = result.format(XMLNS_ROUTE53, health_check.to_xml()).strip()
+        return result
```

Comment:
this is exactly the kind of code we should remove, since it is handled by ASF. could we implement the `get_health_check` API methods in the provider and wrap moto that way, rather than patching it?